native-android/src/lib/notifications.ts

```diff
@@ -0,0 +1,155 @@
+  storage.save({ key: 'hasRequestedNotifications', data: value });
+};
+
+export const initializePushNotifications = async () => {
+  if (!Device.isDevice) {
+    return false;
+  }
+
+  Notifications.setNotificationHandler({
+    handleNotification: async () => ({
+      shouldShowAlert: true,
+      shouldPlaySound: true,
+      shouldSetBadge: false,
+    }),
+  });
+
+  await Notifications.setNotificationCategoryAsync('message', [
+    {
+      identifier: 'reply',
+      buttonTitle: 'Reply',
+      options: {
+        opensAppToForeground: true,
+      },
+    },
+    {
+      identifier: 'dismiss',
+      buttonTitle: 'Dismiss',
+      options: {
+        opensAppToForeground: false,
+      },
+    },
+  ]);
+
+  if (Platform.OS === 'android') {
+    await Notifications.setNotificationChannelAsync('default', {
+      name: 'default',
+      importance: Notifications.AndroidImportance.MAX,
+      vibrationPattern: [0, 250, 250, 250],
+      lightColor: '#FF231F7C',
+    });
+  }
+
+  const hasRequestedNotifications = await getHasRequestedNotifications();
+  console.debug(
+    'Already asked for notifications permission?',
+    hasRequestedNotifications ? 'Yes' : 'No'
+  );
+
+  try {
+    const { status } = await Notifications.getPermissionsAsync();
+    console.debug('Current push notifications setting:', status);
+
+    const isGranted = status === 'granted';
+    if (hasRequestedNotifications || isGranted) {
+      setHasRequestedNotifications(true);
+      return isGranted;
+    }
+  } catch (err) {
+    console.error('Error reading current push notification setting:', err);
+  }
+
+  try {
+    const { status: nextStatus } =
+      await Notifications.requestPermissionsAsync();
+    console.debug('New push notifications setting:', nextStatus);
+
+    setHasRequestedNotifications(true);
+    if (nextStatus !== 'granted') {
+      return false;
+    }
+  } catch (err) {
+    console.error('Error requesting new push notification setting:', err);
+    return false;
+  }
+
+  const { data: token } = await Notifications.getDevicePushTokenAsync();
+  console.debug('Obtained new push notification token:', token);
+  await pokeNotify(token);
+
+  return true;
+};
+
+export const pokeNotify = async (token: string) => {
+  const api = useStore.getState().api;
+  await api?.poke({
+    app: 'notify',
+    mark: 'notify-client-action',
+    json: {
+      'connect-provider-with-binding': {
+        who: NOTIFY_PROVIDER,
+        service: NOTIFY_SERVICE,
+        address: token,
+        binding: Platform.OS === 'android' ? 'fcm' : 'apn',
+      },
+    },
+  });
+};
+
+export const handleNotification = async (
+  notification: Notifications.Notification
+) => {
+  const { identifier, content } = notification.request;
+  console.log('Handling notification', identifier, content.data);
+  await Notifications.scheduleNotificationAsync({
+    identifier,
+    content: {
+      categoryIdentifier: 'message',
+      title: 'You received a new message',
```

Comment:
do we not want to scry for the contents here?

Reply:
We will scry for contents in a future build